arwendebug/world.go

```diff
@@ -0,0 +1,138 @@
+package arwendebug
+
+import (
+	"github.com/ElrondNetwork/arwen-wasm-vm/arwen"
+	"github.com/ElrondNetwork/arwen-wasm-vm/arwen/host"
+	"github.com/ElrondNetwork/arwen-wasm-vm/config"
+	"github.com/ElrondNetwork/arwen-wasm-vm/ipc/arwenpart"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+type worldDataModel struct {
+	ID       string
+	Accounts AccountsMap
+}
+
+type world struct {
+	id             string
+	blockchainHook *BlockchainHookMock
+	vm             vmcommon.VMExecutionHandler
+}
+
+func newWorldDataModel(worldID string) *worldDataModel {
+	return &worldDataModel{
+		ID:       worldID,
+		Accounts: make(AccountsMap),
+	}
+}
+
+// newWorld creates a new debugging world
+func newWorld(dataModel *worldDataModel) (*world, error) {
+	blockchainHook := NewBlockchainHookMock()
+	blockchainHook.Accounts = dataModel.Accounts
+
+	vm, err := host.NewArwenVM(
+		blockchainHook,
+		arwenpart.NewCryptoHookGateway(),
+		getHostParameters(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &world{
+		id:             dataModel.ID,
+		blockchainHook: blockchainHook,
+		vm:             vm,
+	}, nil
+}
+
+func getHostParameters() *arwen.VMHostParameters {
+	return &arwen.VMHostParameters{
+		VMType:                   []byte{5, 0},
+		BlockGasLimit:            uint64(10000000),
+		GasSchedule:              config.MakeGasMap(1, 1),
+		ElrondProtectedKeyPrefix: []byte("ELROND"),
+	}
+}
+
+func (world *world) deploySmartContract(request DeployRequest) (*DeployResponse, error) {
```

Comment:
w *world

Reply:
no need for error in return - as it is not used

Reply:
Renamed; removed error.